node_modules/oae-principals/tests/test-emails.js

```diff
@@ -0,0 +1,603 @@
+            PrincipalsTestUtil.assertCreateUserSucceeds(restContext, params, function(user, oldToken) {
+                RestAPI.Authentication.login(restContext, params.username, 'password', function(err) {
+                    assert.ok(!err);
+
+                    // Resend the email verification token
+                    PrincipalsTestUtil.assertResendEmailTokenSucceeds(restContext, user.id, function(newToken) {
+
+                        // Assert we can still use the old one
+                        PrincipalsTestUtil.assertVerifyEmailSucceeds(restContext, params.email, oldToken, function() {
+                            return callback();
+                        });
+                    });
+                });
+            });
+        });
+    });
+
+    describe('Moments when to verify an email address', function() {
+
+        /**
+         * Test that verifies that local accounts need to verify their email address
+         */
+        it('verify local user accounts need to verify their email address', function(callback) {
+            // We can't use TestsUtil.generateTestUsers as that would do the verification process for us
+            var email = TestsUtil.generateTestEmailAddress();
+            var params = {
+                'username': TestsUtil.generateTestUserId(),
+                'password': 'password',
+                'displayName': TestsUtil.generateRandomText(1),
+                'email': email
+            };
+            PrincipalsTestUtil.assertCreateUserSucceeds(anonymousRestContext, params, function(user, token) {
+
+                // Sanity check the user has no `email` property yet
+                RestAPI.User.getUser(camAdminRestContext, user.id, function(err, user) {
+                    assert.ok(!err);
+                    assert.ok(!user.email);
+                    assert.strictEqual(user.emailVerified, false);
+                    assert.strictEqual(user.emailChange, email.toLowerCase());
+
+                    // Assert that there's no mapping yet for the email address
+                    PrincipalsTestUtil.assertUserEmailMapping(email, [], function() {
+
+                        // Verify the email address
+                        PrincipalsTestUtil.assertVerifyEmailSucceeds(anonymousRestContext, email, token, function() {
+
+                            // Assert the mapping has been created
+                            PrincipalsTestUtil.assertUserEmailMapping(email, [user.id], function() {
+                                return callback();
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that local users accounts created by a tenant administrator do not need to verify their email address
+         */
+        it('verify local user accounts created by a tenant administrator do not need to verify their email address', function(callback) {
```

Comment:
I think they still should, because a tenant administrator could make a mistake typing it out